### DISCUSSION TITLE
Refresh when opening the app

### DISCUSSION THREAD

**Discussion by PhilC813** [breezy-weather/breezy-weather#1402] [Ideas]
### Describe your idea

I know this is very clearly indicated as being something that won't be implemented in the FAQ, but why?

I've spent the last year or so trying out a few weather apps to see which one I like best. Currently trying out this app and Weawow, which, at least if you consider the rating, is considered the undeniable king on Google Play. Their app, and nearly every other I tried, have an automatic refresh once the app is opened. It's a natural, logical behaviour, and it isn't contradictory to make an exception to the background refresh process for this.

I know I probably won't change the mind of head developers here, but just throwing this out there. There's many people that requested it, so it's definitely not a small, niche request.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
It's not a matter of niche request, it's a matter of not making calls when the weather data is still fresh for various reasons: we use free-tier for API so we don't want to uselessly call them, Green IT purposes, etc.

You mentioned the background update process. If you turn it off, it will automatically refresh on opening the location if it's been more than 1.5 hours since last refresh.

Lastly, you can still force that behavior by pulling to refresh, which takes just a second to do, so only users who absolutely want a refresh (despite the weather data still being fresh) do it.

**Reply by PhilC813** [reply]
Thank you for taking the time to reply and explain the logic.

Weather data may still be fresh, but if the user takes the time to open the app, I don't think it's unreasonable to add another refresh. If green IT is a concern, it could simply be added as an option within the settings, and be disabled by default. The Weawow weather app is also free, but it still refreshes upon an app launch.

When you test the different providers when setting Breezy for the first time, you make dozens of API calls with no restriction. I don't think that adding one or two additional API calls a day is a deal-breaker. At least that's how I see it.

Open source apps are known for giving power to the user, and provide customizability, so it's a bit disappointing. I know that I can simply pull down to refresh, but I shouldn't have to, so it becomes an annoyance.

**Reply by papjul** [reply]
If the data is still fresh, why do you need a refresh? The "automatic refresh" logic already exists, but applies when the data is outdated. It's not just one or two API calls a day. If you keep swapping between apps, it will refresh every 10 seconds. It even becomes an annoyance then. That's why we restrict this to only refresh when it's been long enough that the data is outdated.

Additionally, I don't consider a 1 sec gesture one or twice a day an annoyance.

**Reply by PhilC813** [reply]
The weather forecast will still be fresh, but in many cases, not the current weather preview.

It obviously isn't an annoyance to you, but if many users have asked for such a feature, it is to many.

**Reply by PhilC813** [reply]
To compare again to Weawow:
When you switch between apps, their app doesn't trigger a refresh. Switching between recent apps isn't considered an app launch.

**Reply by papjul** [reply]
> The weather forecast will still be fresh, but in many cases, not the current weather preview.

I do pull to refresh for that reason, when I notice it's been a while since it was last refreshed (but it doesn't happen a lot since my personal setup is a 1.5 hours refresh rate).

> When you switch between apps, their app doesn't trigger a refresh. Switching between recent apps isn't considered an app launch.

So, just to get an automatic refresh on opening, you swipe up Weawow from your recent apps everytime you use it, then tap the app again later and wait for it to open? That seems way more an annoyance than just pulling to refresh.

Note that I'm not trying to make you agree with me or switch to Breezy Weather. If you got used to that behavior and you're happy with the Weawow app, that's fine! I'm not trying to implement every other weather app features (hence that long list of features that will not be implemented) ;)
